src/Libraries/Microsoft.Extensions.AI.OpenAI/OpenAIChatClient.cs

```diff
@@ -209,6 +209,10 @@ private static List<ChatMessageContentPart> ToOpenAIChatContent(IList<AIContent>
                     }
 
                     break;
+
+                case DataContent dataContent when dataContent.MediaType.StartsWith("application/pdf", StringComparison.OrdinalIgnoreCase):
+                    parts.Add(ChatMessageContentPart.CreateFilePart(BinaryData.FromBytes(dataContent.Data), dataContent.MediaType, $"{Guid.NewGuid():N}.pdf"));
+                    break;
```

Comment:
Would there be a drawback to just mapping all other `DataContent` objects, not limiting it to PDF? Are there cases where someone would add a `DataContent` but not want it to be sent in the request?

Reply:
Just that, at least today, anything else will result in the request failing.